tests/interpolation/test_interpolation.py

```diff
@@ -444,3 +444,53 @@ def test_type_validation_error_no_throw() -> None:
     cfg = OmegaConf.structured(User(name="Bond", age=SI("${name}")))
     bad_node = cfg._get_node("age")
     assert bad_node._maybe_dereference_node() is None
+
+
+@mark.parametrize(
+    ("cfg", "key"),
+    [
+        param({"a": {"a": "${a}"}}, "a.a"),
+        param({"a": {"a": "${..a}"}}, "a.a"),
+    ],
+)
+def test_parent_interpolation(cfg: Any, key: str) -> None:
+    cfg = OmegaConf.create(cfg)
+    with raises(
+        InterpolationResolutionError,
+        match=re.escape("Interpolation to parent node detected"),
+    ):
+        OmegaConf.select(cfg, key)
+
+
+IRE = InterpolationResolutionError
+
+
+@mark.parametrize(
+    ("cfg", "key", "expected"),
+    [
+        param({"a": "${a}"}, "a", IRE, id="self_interpolation"),
+        param({"a": "${b}", "b": "${a}"}, "a", IRE, id="ping-pong"),
+        param({"a": {"a": "${b}"}, "b": "${a.a}"}, "a.a", IRE, id="ping-pong"),
+        param({"a": {"a": "${.a}"}}, "a.a", IRE, id="self-relative"),
+        # quoted strings
+        param({"a": "${oc.decode:'${a}'}"}, "a", IRE, id="quoted"),
+        # custom resolver
+        param({"a": "${oc.identity:${a}}"}, "a", IRE, id="custom-resolver"),
+        param(
+            {"a": "${oc.identity:${oc.identity:${a}}}"}, "a", IRE, id="custom-resolver"
```

Comment:
Should actually be `identity` and not `oc.identity`. Technically the test will work with any resolver name since the exception will be raised before checking the resolver, but if for any reason the recursion exception isn't raised then it would be an issue (since the unknown resolver name would also trigger an `IRE` and thus the test would still pass).

Reply:
True and good catch.
However, another IRE will not have the message "Recursive interpolation detected", so I am not sure what is going on here.

Reply:
oh, actually I know what is going on here: the recursive interpolations are resolved before calling oc.identity and discovering there is no such thing.

Reply:
> oh, actually I know what is going on here: the recursive interpolations are resolved before calling oc.identity and discovering there is no such thing.

Yes exactly, my concern was what would happen if for any reason there would be no error anymore due to the cycle: then the test would have still passed by failing on the resolver name.
However, I hadn't noticed that the test was actually matching the text of the exception, so it wouldn't actually have been a real issue.